percona/controllers/pgcluster_test.go

```diff
@@ -54,3 +56,150 @@ var _ = Describe("PG Cluster", Ordered, func() {
 		})
 	})
 })
+
+var _ = Describe("PMM sidecar", Ordered, func() {
+	ctx := context.Background()
+
+	const crName = "pmm-test"
+	const ns = crName
+	crNamespacedName := types.NamespacedName{Name: crName, Namespace: ns}
+
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      crName,
+			Namespace: ns,
+		},
+	}
+
+	BeforeAll(func() {
+		By("Creating the Namespace to perform the tests")
+		err := k8sClient.Create(ctx, namespace)
+		Expect(err).To(Not(HaveOccurred()))
+	})
+
+	AfterAll(func() {
+		By("Deleting the Namespace to perform the tests")
+		_ = k8sClient.Delete(ctx, namespace)
+	})
+
+	cr, err := readDefaultCR(crName, ns)
+	It("should read defautl cr.yaml", func() {
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("should create PerconaPGCluster with pmm enabled", func() {
+		cr.Spec.PMM.Enabled = true
+		Expect(k8sClient.Create(ctx, cr)).Should(Succeed())
+	})
+
+	Context("Reconcile controller", func() {
+		It("Controller should reconcile", func() {
+			_, err := reconciler().Reconcile(ctx, ctrl.Request{NamespacedName: crNamespacedName})
+			Expect(err).NotTo(HaveOccurred())
+			_, err = crunchyReconciler().Reconcile(ctx, ctrl.Request{NamespacedName: crNamespacedName})
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	sts := appsv1.StatefulSet{}
+	Context("sts should not have pmm sidecar", func() {
```

Comment:
From this description, I can't quite get the context you're testing here. Be a bit more descriptive with the rest of the `Contexts` as well. `Context` has an alias `When` so that could make it a bit more clear as well, `When(something is in place) --> then It(should be such and such)`. `Context` blocks in general should not contain `should`, `It` blocks are for that. 
Also, by Gingko convention container nodes (Context, When) start in uppercase, and specs (It, Specify) start in lowercase.

More info: https://onsi.github.io/ginkgo/#spec-subjects-it

Reply:
https://github.com/percona/percona-postgresql-operator/pull/394/commits/58cb784cb7832eda9e81ab10429926eb6cce167e